HousingRepairsSchedulingApi/Startup.cs

```diff
@@ -66,5 +90,46 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 endpoints.MapControllers().RequireAuthorization();
             });
         }
+
+        private void ConfigureOptions(IServiceCollection services)
+        {
+            var drsOptionsConfiguration = this.Configuration.GetSection(nameof(DrsOptions));
+
+            if (string.IsNullOrEmpty(drsOptionsConfiguration["ApiAddress"]))
+            {
+                throw new InvalidOperationException($"Incorrect configuration: {nameof(DrsOptions)}.{nameof(DrsOptions.ApiAddress)} is a required configuration.");
+            }
+            if (string.IsNullOrEmpty(drsOptionsConfiguration["Login"]))
+            {
+                throw new InvalidOperationException($"Incorrect configuration: {nameof(DrsOptions)}.{nameof(DrsOptions.Login)} is a required configuration.");
+            }
+            if (string.IsNullOrEmpty(drsOptionsConfiguration["Password"]))
+            {
+                throw new InvalidOperationException($"Incorrect configuration: {nameof(DrsOptions)}.{nameof(DrsOptions.Password)} is a required configuration.");
+            }
+
+            services.Configure<DrsOptions>(drsOptionsConfiguration);
+        }
+
+        private static HttpBindingBase CreateBinding(Uri uri)
+        {
+            HttpBindingBase binding;
+            var uriScheme = uri.Scheme;
+
+            if (uriScheme == Uri.UriSchemeHttps)
```

Comment:
Couldn't use a `switch` here due to the schemes under `Uri` not being constants (they are `readonly` instead.)